### DISCUSSION TITLE
About training transforms.Normalize

### DISCUSSION THREAD

**Discussion by crapthings** [huggingface/diffusers#9805] [General]
@sayakpaul 

is this same thing when training?

transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
transforms.Normalize([0.5], [0.5])
transforms.Normalize([0.5, 0.5, 0.5], [0.5, 0.5, 0.5])

what is different on result image?

**Comment by sayakpaul** [comment]
`transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))`
This normalization is designed for 3-channel images (like RGB).

`transforms.Normalize([0.5], [0.5])`
This is meant for single-channel images (like grayscale).

`transforms.Normalize([0.5, 0.5, 0.5], [0.5, 0.5, 0.5])`
Same as first.

**Reply by crapthings** [reply]
so i see all training code use this one

transforms.Normalize([0.5], [0.5])
This is meant for single-channel images (like grayscale).


what is different on inference result between two after training with single channel vs rgb channel?

https://github.com/huggingface/diffusers/blob/main/examples/dreambooth/train_dreambooth.py#L664
https://github.com/huggingface/diffusers/blob/main/examples/controlnet/train_controlnet.py#L687

**Reply by crapthings** [reply]
@sayakpaul if inference code doesn't denorm will that make final image color offset?

**Reply by sayakpaul** [reply]
The inference code doesn't have to denorm no? We don't use the VAE encoder during inference. What am I missing?
